CharlieBackend.Core/Models/ResultModel/Result.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace CharlieBackend.Core.Models.ResultModel
+{
+    public class Result<T>
+    {
+
+        public Task<T> TransferredData { get; private set; }
+
+        public Error Error { get; private set; }
+
+        public static Task<Result<T>> ReturnData(Task<T> transferredData)
```

Comment:
1. I think it would be better to name it Success
2. This method can be completely sync, so please omit Task<>